src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -142,9 +142,9 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 var completionItem = new TCompletionItem
                 {
                     Label = completeDisplayText,
-                    InsertText = item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : completeDisplayText,
-                    SortText = item.SortText,
-                    FilterText = item.FilterText,
+                    InsertText = (item.Properties.ContainsKey("InsertionText") && !item.Properties["InsertionText"].Equals(item.DisplayText, StringComparison.Ordinal)) ? item.Properties["InsertionText"] : null,
+                    SortText = (item.HasDifferentSortText) ? item.SortText : null,
+                    FilterText = (item.HasDifferentFilterText) ? item.FilterText : null,
```

Comment:
I believe this also needs to be comparing against the label since that is the fallback in LSP